internal/transport/transport_test.go

```diff
@@ -2746,3 +2746,100 @@ func (s) TestClientSendsAGoAwayFrame(t *testing.T) {
 		t.Errorf("Context timed out")
 	}
 }
+
+// Test that in the event of a graceful client transport shutdown
+// , i.e., clientTransport.Close(), if the conn hung for
+// LoopyWriterTimeout, client should still be close itself and should
+// not wait for long.
+func (s) TestClientCloseReturnsEarlyWhenGoAwayWriteHangs(t *testing.T) {
+	// Override goAwayLoopyWriterTimeout to 0 so that we always
+	// time out while writing GOAWAY on client.Close(). This is
```

Comment:
suggestion: override timer for writing GOAWAY to 0 so that the conn write always times out. It is equivalent of real network hang when conn write for goaway doesn't finish in specified deadline